src/session/sidebar/avatar.rs

```diff
@@ -176,7 +177,21 @@ impl Avatar {
             binding.unwatch();
         }
 
-        self.imp().avatar.set_item(item.clone());
+        if let Some(chat) = item.and_downcast_ref::<Chat>() {
+            if chat.is_own_chat() {
+                self.imp().avatar.set_item(item.clone());
+            } else {
+                match chat.type_() {
+                    ChatType::Private(user) => {
+                        imp.avatar.set_item(Some(user.clone().upcast()));
+                    }
+                    ChatType::Secret(secret) => {
+                        imp.avatar.set_item(Some(secret.user().clone().upcast()));
+                    }
+                    _ => self.imp().avatar.set_item(item.clone()),
+                }
+            }
+        }
```

Comment:
1. This should rather be merged with the `if let` in line 196.
2. What about a pure User? You don't set it as an avatar item.

Reply:
What do you mean pure `User`?
Are you referring to own user?

Reply:
You don't call `imp.avatar.set_item` for an item that is a pure `User`, but just for a `Chat` or the User of a chat.
Line 28 states that `item` can be a Chat or a User
```rust
        /// A `Chat` or `User`
        pub(super) item: RefCell<Option<glib::Object>>,
```

Reply:
Okay I get it, but what's the problem? Is there any difference between `User` and `User` of the Chat?

It's implemented in the similar way in message_row
```rust 
let avatar_item = match message.sender() {
    MessageSender::User(user) => user.clone().upcast(),
    MessageSender::Chat(chat) => chat.clone().upcast(),
};
avatar.set_custom_text(None);
avatar.set_item(Some(avatar_item));
```

Reply:
Whenever `item` is a `User` it won't have an avatar item set.

Reply:
Sorry, sometimes I'm dumb. I thought it was a problem with the highlighted code. Initially I haven't considered that the item can be User and I still have no idea when the item is User, but I must miss something if the specs says so.